### DISCUSSION TITLE
Feature Request: Add "delete safety" for entries that have existing relations

### DISCUSSION THREAD

**Discussion by jakedohm** [craftcms/cms#8629] [Ideas]
### Description

**Scenario**

I delete an entry from the CP, and that entry is selected in an entry field somewhere

- Current behavior: I'm able to delete the blog post
- Requested behavior: I am *at least* warned that the entry is relied on in a relation.

**The main problem this solves**:

If I set an entries field to be required, when templating I should be able to assume that an entry (or multiple) will be selected, and not have to defensively template against that entries field being empty.

This isn't currently true, as an entry (or asset) could be deleted and that required field left empty.

**Thoughts**

Maybe you could make a distinction between required/non-required entries fields, because of the problem mentioned above. If an entries field is required, you should be able to rely on it. If it isn't, you already know that it might be empty.

**Configuration**
 I think this feature has a few opportunities for configuration:
- On/off - some users might not want this feature at all
- User Permission: Can this user (or group) delete when there is a (required?) relation
- More that I can't think of right now 😂

**Comment by KatieMFritz** [comment]
Nice idea!! It could also be useful to know whether an asset is being referenced by entries or other elements. I'm using various plugins now to show relationships between elements, but I think an out-of-the-box delete warning when something is being used would save many headaches.
